homeassistant/components/sensor/mopar.py

```diff
@@ -0,0 +1,166 @@
+SERVICE_REMOTE_COMMAND = 'remote_command'
+COOKIE = 'mopar_cookies.pickle'
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_USERNAME): cv.string,
+    vol.Required(CONF_PASSWORD): cv.string,
+    vol.Required(CONF_PIN): cv.positive_int
+})
+
+REMOTE_COMMAND_SCHEMA = vol.Schema({
+    vol.Required(ATTR_COMMAND): cv.string,
+    vol.Required(ATTR_VEHICLE_INDEX): cv.positive_int
+})
+
+
+# pylint: disable=unused-argument
+def setup_platform(hass, config, add_devices, discovery_info=None):
+    """Setup the Mopar platform."""
+    import motorparts
+    cookie = hass.config.path(COOKIE)
+    try:
+        session = motorparts.get_session(config.get(CONF_USERNAME),
+                                         config.get(CONF_PASSWORD),
+                                         config.get(CONF_PIN),
+                                         cookie_path=cookie)
+    except motorparts.MoparError:
+        _LOGGER.error("failed to login")
+        return False
+
+    def _handle_service(service):
+        """Handle service call."""
+        index = service.data.get(ATTR_VEHICLE_INDEX)
+        command = service.data.get(ATTR_COMMAND)
+        try:
+            motorparts.remote_command(session, command, index)
+        except motorparts.MoparError as error:
+            _LOGGER.error(str(error))
+
+    hass.services.register(DOMAIN, SERVICE_REMOTE_COMMAND, _handle_service,
+                           schema=REMOTE_COMMAND_SCHEMA)
+
+    hass.data[DATA_MOPAR] = MoparData(session)
+    add_devices([MoparSensor(hass, index)
+                 for index, _ in enumerate(hass.data[DATA_MOPAR].vehicles)],
+                True)
+    return True
+
+
+# pylint: disable=too-few-public-methods
+class MoparData(object):
+    """Container for Mopar vehicle data.
+
+    Prevents session expiry re-login race condition.
+    """
+
+    def __init__(self, session):
+        """Initialize data."""
+        self._session = session
+        self.vehicles = []
+        self.vhrs = {}
+        self.tow_guides = {}
+        self.update()
+
+    @Throttle(MIN_TIME_BETWEEN_UPDATES)
+    def update(self, **kwargs):
+        """Update data."""
+        import motorparts
+        _LOGGER.info("updating vehicle data")
+        try:
+            self.vehicles = motorparts.get_summary(self._session)['vehicles']
+        except motorparts.MoparError:
+            _LOGGER.exception("failed to get summary")
+            return
+        for index, _ in enumerate(self.vehicles):
+            try:
+                self.vhrs[index] = motorparts.get_report(self._session, index)
+                self.tow_guides[index] = motorparts.get_tow_guide(
+                    self._session, index)
+            except motorparts.MoparError:
+                _LOGGER.warning("failed to update for vehicle index %s", index)
+
+
+class MoparSensor(Entity):
+    """Mopar vehicle sensor."""
+
+    def __init__(self, hass, index):
+        """Initialize the sensor."""
+        self._hass = hass
```

Comment:
Don't pass hass to contstructor. You can use `self.hass` from Entity inside this object.

Reply:
```
...
    self._data = self.hass.data[DATA_MOPAR]
AttributeError: 'NoneType' object has no attribute 'data'
```

Do I need to do anything else? I see `hass` is initialized but not set in `class Entity`.

Reply:
You are correct. Pass the object with init parameter